internal/abci/abci.go

```diff
@@ -1450,7 +1483,36 @@ func (a *AbciApp) Query(ctx context.Context, req *abciTypes.RequestQuery) (*abci
 			return nil, err
 		}
 		return &abciTypes.ResponseQuery{Value: bts}, nil
+	} else if strings.HasPrefix(req.Path, ABCIPeerFilterPath) {
+		// When CometBFT connects to a peer, it sends two queries to the ABCI application
+		// using the following paths, with no additional data:
+		//   - `/p2p/filter/addr/<IP:PORT>`
+		// 	 - `p2p/filter/id/<ID>` where ID is the peer's node ID
+		// If either of these queries return a non-zero ABCI code, CometBFT will refuse to connect to the peer.
+		// We manage allowed list based on the peer's node ID rather than the IP addresses, so we only need to
+		// handle the `id` query and return OK for all `addr` queries.
+
+		paths := strings.Split(req.Path[ABCIPeerFilterPathLen:], "/")
+		if len(paths) != 2 {
+			return &abciTypes.ResponseQuery{Code: 1}, fmt.Errorf("invalid path: %s", req.Path)
+		}
+
+		switch paths[0] {
+		case "id":
+			if a.p2p.IsPeerWhitelisted(paths[1]) {
+				return &abciTypes.ResponseQuery{Code: abciTypes.CodeTypeOK}, nil
```

Comment:
You had a log here letting the operator know that the connection was successful. Curious why you took it out?

Reply:
felt kinda unnecessary, as cometbft p2p layer would log the reason if we reject the request